Automattic/wp-calypso
# Move and modularise guided tour state

This PR is one of many working on modularising state in Calypso. This one handles guided tours, which until now live under `ui` but deserve their own top-level entry.

For more details on state modularisation, see the [modularised state documentation](https://github.com/Automattic/wp-calypso/blob/master/docs/modularized-state.md) and p4TIVU-9lM-p2.

Note: I added all the reviewers that GitHub suggested. Please feel free to ignore the review request or pull in someone else if you're not comfortable reviewing this PR. Thank you!

#### Changes proposed in this Pull Request

* Move guided tours state out of `ui` and into a new top level entry
* Modularise it
* Split `guided-tours/contexts` monolithic module into smaller ones.

#### Testing instructions

Guided tour state is always loaded on (logged-in) boot, and that appears to be an intentional architectural decision. However, that makes it difficult to see the state going from unloaded to loaded, as it happens without user input on boot, so it's difficult to test the automatic state loading part of the PR. That's not a big concern, however; the important thing is to ensure that nothing is getting broken.

With that in mind, in order to verify that this change works correctly, please smoke test guided tour functionality and ensure that it continues to work as before.
